Google.PowerShell/Common/GcloudCmdlet.cs

```diff
@@ -418,5 +419,52 @@ public static string GetUriPart(string resourceType, string uri)
             Match match = Regex.Match(uri, $"{resourceType}/(?<value>[^/]*)");
             return match.Groups["value"].Value;
         }
+
+        /// <summary>
+        /// Generate a RuntimeDefinedParameter based on the parameter name,
+        /// the help message and the valid set of parameter values.
+        /// </summary>
+        public RuntimeDefinedParameter GenerateRuntimeParameter(
+            string parameterName,
+            string helpMessage,
+            string[] validSet,
+            bool isMandatory = false,
+            params string[] parameterSetNames)
+        {
+            List<Attribute> attributeLists = new List<Attribute>();
+
+            if (parameterSetNames.Length == 0)
+            {
+                ParameterAttribute paramAttribute = new ParameterAttribute()
+                {
+                    Mandatory = isMandatory,
+                    HelpMessage = helpMessage
+                };
+                attributeLists.Add(paramAttribute);
+            }
+            else
+            {
+                for (int i = 0; i < parameterSetNames.Length; i += 1)
+                {
+                    ParameterAttribute paramAttribute = new ParameterAttribute()
+                    {
+                        Mandatory = isMandatory,
+                        HelpMessage = helpMessage
+                    };
+                    paramAttribute.ParameterSetName = parameterSetNames[i];
+                    attributeLists.Add(paramAttribute);
+                }
+            }
+
+            if (validSet.Length != 0)
```

Comment:
Should this be `validSet?.Length > 0` so even a null validSet does the right thing?